### PR TITLE
[dotnet] Added Deprecation to WebElement.GetAttribute() per #13334

### PR DESCRIPTION
**Pull request by shbenzer**
### Description
Added Deprecation to WebElement.GetAttribute() per #13334

### Motivation and Context
#13334 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

___

### **PR Type**
enhancement

___

### **Description**
- Added a deprecation warning to the `GetAttribute` method in the `IWebElement` interface.
- Suggested using the `GetDomAttribute` method as an alternative.

___

PRDescriptionHeader.CHANGES_WALKTHROUGH
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>IWebElement.cs</strong><dd><code>Deprecate `GetAttribute` method in `IWebElement` interface</code></dd></summary>
<hr>

dotnet/src/webdriver/IWebElement.cs

<li>Added an <code>Obsolete</code> attribute to the <code>GetAttribute</code> method.<br> <li> Provided a deprecation warning message suggesting the use of <br><code>GetDomAttribute</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14676/files#diff-09553d2da00fdead17001e2a7558cba418b6e926879880eedf0ed487164055cf">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by nvborisenko** [comment]
```csharp
driver.Url = "https://nuget.org";

var input = driver.FindElement(By.Id("search"));
input.SendKeys("abc");

var val = input.GetAttribute("value");
var val2 = input.GetDomAttribute("value");
var val3 = input.GetDomProperty("value");

Console.WriteLine(val); // "abc"
Console.WriteLine(val2); // ""
Console.WriteLine(val3); // abc
```

**Comment by shbenzer** [comment]
> ```cs
> driver.Url = "https://nuget.org";
> 
> var input = driver.FindElement(By.Id("search"));
> input.SendKeys("abc");
> 
> var val = input.GetAttribute("value");
> var val2 = input.GetDomAttribute("value");
> var val3 = input.GetDomProperty("value");
> 
> Console.WriteLine(val); // "abc"
> Console.WriteLine(val2); // ""
> Console.WriteLine(val3); // abc
> ```

good catch, I forgot to add GetDomProperty() to the deprecation note

**Comment by nvborisenko** [comment]
We have to double check which one is correct: `GetAttribute` or `GetDomAttribute`. I though the correct one is `GetAttribute`.

**Comment by shbenzer** [comment]
> We have to double check which one is correct: `GetAttribute` or `GetDomAttribute`. I though the correct one is `GetAttribute`.

According to @titusfortner in #13334 

>. getAttribute() assumes that people don't know the difference between a property and an attribute and don't want to be bothered to figure it out (a decent assumption), so it uses a bunch of fancy JavaScript that does a pretty good job of guessing whether the user is looking for the property or the attribute and returning that value. It's great convenience, but it is now implemented with a client side atom which means a large payload gets sent to the driver/server on every usage.

> We decided in TLC Meeting 1/18 that we would go ahead and mark this feature deprecated to encourage people to move to the more precise property and attribute methods.

**Comment by nvborisenko** [comment]
All right, `GetAttribute()` is deprecated.

Since it is breaking change of core (most used interface) functionality, let's add info when it will be completely removed. In v5 or in v6? It should be definitely major selenium version, but which one... Usually we remove deprecated methods in 2 minor releases, but this another case, this would be real pain for all users. So major release should be considered. But if v5 will be released in 2 months, then it is not enough time range - then v6 is preferable.

**Comment by shbenzer** [comment]
I agree that major releases should be considered. Since v5 is too soon, I'll put v6 in. 

Worst case we make a change later and make it v7.

File: dotnet/src/webdriver/IWebElement.cs [hunk 1]
```diff
@@ -170,7 +170,7 @@ public interface IWebElement : ISearchContext
         /// </list>
         /// </remarks>
         /// <exception cref="StaleElementReferenceException">Thrown when the target element is no longer valid in the document DOM.</exception>
-        [Obsolete("Use GetDomAttribute(string attributeName) or GetDomProperty(string propertyName")]
+        [Obsolete("Use GetDomAttribute(string attributeName) or GetDomProperty(string propertyName). GetAttribute() is schedule for removal in Selenium v6"]
```
Comment (nvborisenko) [inline dotnet/src/webdriver/IWebElement.cs]: `GetAttribute(string attributeName) will be removed in Selenium 6.` - full signature and typo.

Comment (shbenzer) [reply dotnet/src/webdriver/IWebElement.cs]: Done

**Review by nvborisenko — APPROVED** [review]

**Comment by nvborisenko** [comment]
@shbenzer please do not merge! Kind of code-freeze after release. Keep it open for a while.

**Comment by nvborisenko** [comment]
@shbenzer please check failing CI build, easy to fix.

**Comment by shbenzer** [comment]
seemed to be missing the system import as well

**Pull request merged**